src/props.animations.js

```diff
@@ -9,57 +9,57 @@ export default {
 @keyframes fade-out {
   to { opacity: 0 }
 }`,
-  "--animation-scale-up": "scale-up .5s var(--ease-3)", 
+  "--animation-scale-up": "scale-up var(--scale-duration, 1ms) var(--ease-3)", 
   "--animation-scale-up-@": `
 @keyframes scale-up {
   to { transform: scale(1.25) }
 }`,
-  "--animation-scale-down": "scale-down .5s var(--ease-3)", 
+  "--animation-scale-down": "scale-down var(--scale-duration, 1ms) var(--ease-3)", 
   "--animation-scale-down-@": `
 @keyframes scale-down {
   to { transform: scale(.75) }
 }`,
-  "--animation-slide-out-up": "slide-out-up .5s var(--ease-3)", 
+  "--animation-slide-out-up": "slide-out-up var(--slide-duration, 1ms) var(--ease-3)", 
   "--animation-slide-out-up-@": `
 @keyframes slide-out-up {
   to { transform: translateY(-100%) }
 }`,
-  "--animation-slide-out-down": "slide-out-down .5s var(--ease-3)", 
+  "--animation-slide-out-down": "slide-out-down var(--slide-duration, 1ms) var(--ease-3)", 
   "--animation-slide-out-down-@": `
 @keyframes slide-out-down {
   to { transform: translateY(100%) }
 }`,
-  "--animation-slide-out-right": "slide-out-right .5s var(--ease-3)", 
+  "--animation-slide-out-right": "slide-out-right var(--slide-duration, 1ms) var(--ease-3)", 
   "--animation-slide-out-right-@": `
 @keyframes slide-out-right {
   to { transform: translateX(100%) }
 }`,
-  "--animation-slide-out-left": "slide-out-left .5s var(--ease-3)", 
+  "--animation-slide-out-left": "slide-out-left var(--slide-duration, 1ms) var(--ease-3)", 
   "--animation-slide-out-left-@": `
 @keyframes slide-out-left {
   to { transform: translateX(-100%) }
 }`,
-  "--animation-slide-in-up": "slide-in-up .5s var(--ease-3)", 
+  "--animation-slide-in-up": "slide-in-up var(--slide-duration, 1ms) var(--ease-3)", 
   "--animation-slide-in-up-@": `
 @keyframes slide-in-up {
   from { transform: translateY(100%) }
 }`,
-  "--animation-slide-in-down": "slide-in-down .5s var(--ease-3)", 
+  "--animation-slide-in-down": "slide-in-down var(--slide-duration, 1ms) var(--ease-3)", 
   "--animation-slide-in-down-@": `
 @keyframes slide-in-down {
   from { transform: translateY(-100%) }
 }`,
-  "--animation-slide-in-right": "slide-in-right .5s var(--ease-3)", 
+  "--animation-slide-in-right": "slide-in-right var(--slide-duration, 1ms) var(--ease-3)", 
   "--animation-slide-in-right-@": `
 @keyframes slide-in-right {
   from { transform: translateX(-100%) }
 }`,
-  "--animation-slide-in-left": "slide-in-left .5s var(--ease-3)", 
+  "--animation-slide-in-left": "slide-in-left var(--slide-duration, 1ms) var(--ease-3)", 
```

Comment:
havent looked into the solution yet, but these `slide-in` animations dont looks right with reduced motion turned on. we can definitely have them behave more like `slide-out` animations, they're nice with reduced motion. can you peek into these ones? 🙂

Reply:
Just checked. I agree with you they don't look right, but they are indeed right.

As `slide in`'s `from` state is not on the same spot as the element itself it looks kind of broken. It does basically the same as the `slide-out`'s just the other way around. As the play JavaScript removes the animation only after a while the ` slide-out`'s stays where it is before jumping back, the ` slide-in` does the same by staying on the spot where it started. We don't see the flickering from the `1ms` on the slide-out as it stays where it is upon starting the animation.

The flickering would be removed when setting the animation timing for reduced motion to `0ms`. I wrote a longer explanaition / question regarding this topic inside of discord.